config/routes.rb

```diff
@@ -12,7 +12,23 @@
   get '/users/:user_id/posts/:id/delete' => 'posts#delete' #delete post
   get '/users/:user_id/posts/' => 'posts#index'
   get '/users/:user_id/:user_id/posts/' => 'posts#index'
+  get '/api/users/:user_id/:user_id/posts/' => 'posts#index' #API posts index
   get '/users/:user_id/posts/:id/new_comment' => 'comments#new_comment' #new comment page
   get '/users/:user_id/posts/:id/comment/:comment_id/delete' => 'comments#delete' #delete comment
   get '/users/:user_id/posts/:id/new_like' => 'likes#new_like' #new like
+  namespace :api do
+    namespace :v1 do
+      resources :users, only: %i[show] do
+        resources :posts, only: %i[index] do
+          resources :comments, only: %i[create index]
+        end
+      end
+    end
+  end
+
+  resources :users, only: [] do
+    member do
+      get 'apitoken'
+    end
+  end
```

Comment:
- [ ] There are so many duplication routes. Please make sure to use restful routes and avoid making different URL for the same route.

Reply:
@Dipeshtwis  Removed duplicate routes and cleaned them up to make them more streamlined and efficient